src/sys/fuchsia/eventedfd.rs

```diff
@@ -0,0 +1,246 @@
+use {io, poll, Evented, Ready, Poll, PollOpt, Token};
+use libc;
+use magenta;
+use magenta::HandleBase;
+use sys::fuchsia::{DontDrop, poll_opts_to_wait_async, sys};
+use std::mem;
+use std::os::unix::io::RawFd;
+use std::sync::{Arc, Mutex};
+
+/// Properties of an `EventedFd`'s current registration
+#[derive(Debug)]
+pub struct EventedFdRegistration {
+    pub token: Token,
+    pub (in sys::fuchsia) handle: DontDrop<magenta::Handle>,
+    pub rereg_signals: Option<(magenta::Signals, magenta::WaitAsyncOpts)>,
+}
+
+impl EventedFdRegistration {
+    unsafe fn new(token: Token,
+                  raw_handle: sys::mx_handle_t,
+                  opts: PollOpt,
+                  signals: magenta::Signals) -> Self
+    {
+        let needs_rereg = opts.is_level() && !opts.is_oneshot();
+
+        EventedFdRegistration {
+            token: token,
+            handle: DontDrop::new(magenta::Handle::from_raw(raw_handle)),
+            rereg_signals: if needs_rereg {
+                Some((signals, poll_opts_to_wait_async(opts)))
+            } else {
+                None
+            },
+        }
+    }
+}
+
+/// An event-ed file descriptor. The file descriptor is owned by this structure.
+#[derive(Debug)]
+pub struct EventedFdInner {
+    /// Properties of the current registration.
+    pub registration: Mutex<Option<EventedFdRegistration>>,
```

Comment:
Similar to above, could the fields here be private with public accessors for what's needed?